src/components/KToaster/KToaster.vue

```diff
@@ -41,76 +61,128 @@ defineProps({
 defineEmits<{
   (e: 'close', val: any): void
 }>()
+
+const getToastIcon = (appearance?: ToasterAppearance): ToastIcon => {
+  switch (appearance) {
+    case 'info':
+      return InfoIcon
+    case 'success':
+      return CheckCircleIcon
+    case 'warning':
+      return WarningIcon
+    case 'danger':
+      return ClearIcon
+    case 'system':
+      return KongIcon
+    default:
+      return InfoIcon // info as default in case of invalid appearance
+  }
+}
 </script>
 
 <style lang="scss" scoped>
-@import '@/styles/tmp-variables';
-@import '@/styles/styles';
-
-$transition: all .3s;
-
-.toaster-container-outer {
+.k-toaster {
   bottom: 16px;
-  max-width: 300px;
+  display: flex;
+  flex-direction: column;
+  gap: var(--kui-space-50, $kui-space-50);
   position: fixed;
   right: 16px;
-  transition: $transition;
-  width: auto;
+  width: 400px;
   z-index: 10000;
-}
 
-.toaster-item {
-  box-shadow: 0 0 12px rgba(0,0,0,.12);
-  display: flex;
-  margin-bottom: var(--kui-space-60, $kui-space-60);
-  overflow: hidden;
-  transition: $transition;
-  width: 100%;
-
-  :deep(.k-alert) {
-    background-color: var(--kui-color-background, $kui-color-background);
-    color: var(--kui-color-text, $kui-color-text);
+  .toast {
+    background-color: var(--kui-color-background-inverse, $kui-color-background-inverse);
+    border-radius: var(--kui-border-radius-20, $kui-border-radius-20);
+    box-shadow: var(--kui-shadow, $kui-shadow);
+    color: var(--kui-color-text-inverse, $kui-color-text-inverse);
     display: flex;
-    flex: 1;
-    justify-content: space-between;
-    margin-bottom: var(--kui-space-0, $kui-space-0);
-    padding: var(--kui-space-60, $kui-space-60);
-    text-align: left;
+    flex-direction: column;
+    gap: var(--kui-space-30, $kui-space-30);
+    padding: var(--kui-space-50, $kui-space-50);
+    width: 100%;
+
+    .toast-header {
+      align-items: center;
+      display: flex;
+      gap: var(--kui-space-50, $kui-space-50);
+
+      .toast-icon-container {
+        align-items: center;
+        background-color: var(--kui-color-background-primary-weak, $kui-color-background-primary-weak); // info appearance as default in case of invalid appearance
+        border-radius: var(--kui-border-radius-circle, $kui-border-radius-circle);
+        display: flex;
+        height: 32px;
+        justify-content: center;
+        width: 32px;
+      }
+
+      .toast-title {
+        @include truncate;
```

Comment:
issue: i'm not sure it makes sense to truncate the toast title, instead of wrapping the title text. If we truncate, there is literally no way for the user to see the toast title content.

Reply:
We discussed it with @two24studios in the Slack thread - I suggest we truncate the title when it’s too long so it doesn’t take too much space - this way we limit title to just 3-4 words. If that’s not enough to say whatever the notification needs to say - the longer message should be passed though `message` property.

Reply:
I understand what you're saying, but I don't think it makes sense to truncate does it? we're hiding something the user has no other way of seeing. Wouldn't it be the responsibility of the host app to pass in a reasonable string?

Reply:
Well yeah I guess that's what I meant, that it would be host app's responsibility to make sure to provide reasonable length string and compliment it with a message when needed. We can also add `title` attribute as a way for users to view the full title on hover.

Reply:
you can't really expect a user to hover on a tooltip that goes away

Reply:
I'll resolve the thread; just still not sure this is a great pattern